--- /media/maousi/Data/Documents/Programmation/git/VAD-0D-DNN/modelica/parameters/ModelParametersSHF	Tue Mar  3 22:22:17 2020
+++ /media/maousi/Data/Documents/Programmation/git/VAD-0D-DNN/modelica/parameters/ModelParametersSHF8000	Tue Mar  3 22:22:48 2020
@@ -1,4 +1,4 @@
-model ModelParametersSHF
+model ModelParametersSHF8000
   parameter Real Param_LeftAtrium_C = 19.23;
   parameter Real Param_LeftAtrium_V0(unit = "ml") = 25;
   parameter Real Param_LeftAtrium_Vu0(unit = "ml") = 25;
@@ -116,4 +116,4 @@
   parameter Real Param_AutoregulationCenter_fcs0 = 25;
   parameter Real Param_AutoregulationCenter_kev = 7.06;
   parameter Real Param_LVAD_RPM(unit = "RPM") = 8000;
-end ModelParametersSHF;
+end ModelParametersSHF8000;
